### ISSUE TITLE
Slider: prevents enter "-" in associated <p:spinner> component

### ISSUE THREAD

**Issue by skybber** [primefaces/primefaces#6319]
1) Environment

PrimeFaces version: 8.0

2) Expected behavior

Allow enter of "-" in associated input if `minValue` is less then 0

3) Actual behavior

Slider prevents enter "-" in associated input.

probably skipped by this line:

https://github.com/primefaces/primefaces/blob/master/src/main/resources/META-INF/resources/primefaces/slider/slider.js#L127

4) Example XHTML

```xml
  <p:spinner value="#{bean.value}"
        min="-10"
        max="10"
  />
  <p:slider for="@previous" 
        minValue="-10"
        maxValue="10"
  />
```

**Comment by melloware** [comment]
Reproducer proving the issue:
[pf-6319.zip](https://github.com/primefaces/primefaces/files/5224668/pf-6319.zip)

**Comment by melloware** [comment]
Thanks for reporting this will be fixed for 9.0 but if you need it right now just add this MonkeyPatch JS to your app.
```javascript
PrimeFaces.widget.Slider.prototype.bindEvents = function() {
    var $this = this;

    this.jq.on('slide', function(event, ui) {
        $this.onSlide(event, ui);
    });

    if (this.cfg.onSlideStart) {
        this.jq.on('slidestart', function(event, ui) {
            $this.cfg.onSlideStart.call(this, event, ui);
        });
    }

    this.jq.on('slidestop', function(event, ui) {
        $this.onSlideEnd(event, ui);
    });

    if (this.input.parent().hasClass('ui-inputnumber')) {
        this.input.parent().find('input:hidden').on('change', function() {
            $this.setValue($(this).val());
        });
    } else {
        this.input.on('keydown.slider', function(e) {
            var keyCode = $.ui.keyCode,
                key = e.which;

            switch (key) {
                case keyCode.UP:
                case keyCode.DOWN:
                case keyCode.LEFT:
                case keyCode.RIGHT:
                case keyCode.BACKSPACE:
                case keyCode.DELETE:
                case keyCode.END:
                case keyCode.HOME:
                case keyCode.TAB:
                    break;

                default:
                    var metaKey = e.metaKey || e.ctrlKey,
                        isNumber = (key >= 48 && key <= 57) || (key >= 96 && key <= 105) || (key === 190);

                    //prevent special characters with alt and shift
                    if (e.altKey || (e.shiftKey && !(key === keyCode.UP || key === keyCode.DOWN || key === keyCode.LEFT || key === keyCode.RIGHT))) {
                        e.preventDefault();
                    }

                    //prevent letters and allow letters with meta key such as ctrl+c
                    //also allow '.' when the step is defined as decimal
                    // #6319 if number allow negative sign
                    if (!isNumber && !metaKey &&
                        !($this.cfg.min < 0 && key === 189) &&
                        !($this.decimalStep && key === 190)) {
                        e.preventDefault();
                    }
                    break;
            }
        }).on('keyup.slider', function(e) {
            $this.setValue($this.input.val());
        });
    }
};
```
